File: Python/Codes/Dosen.py
"""
Saya Villeneuve Andhira Suwandhi NIP 2108067 mengerjakan Latihan 4
dalam mata kuliah Desain Pemrograman Berorientasi Objek untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin.
"""

# Import #
from SivitasAkademik import SivitasAkademik

# Class declaration #
class Dosen(SivitasAkademik):
    # Private class #
    __employeeID = ""               # NIP
    __lastEdu = ""                  # pend_terakhir
    __skill = ""                    # keahlian

    # default constructor #
    def __init__(self, NIK = "", nama = "", jenis_kelamin = "", email_edu = "", asal_universitas = "", fakultas = "", NIP = "", pend_terakhir = "", keahlian = ""):
        super().__init__(NIK, nama, jenis_kelamin, email_edu, asal_universitas, fakultas)
        self.__employeeID = NIP
        self.__lastEdu = pend_terakhir
        self.__skill = keahlian
    
    # setter #
    def setEmployeeID(self, NIP):
        self.__employeeID = NIP
    def setLastEdu(self, pend_terakhir):
        self.__lastEdu = pend_terakhir
    def setSkill(self, keahlian):
        self.__skill = keahlian
    
    # getter #
    def getEmployeeID(self):
        return self.__employeeID
    def getLastEdu(self):
        return self.__lastEdu
    def getSkill(self):
        return self.__skill

File: C++/Program/Dosen.cpp
/*
Saya Villeneuve Andhira Suwandhi NIM 2108067 mengerjakan Latihan 4
dalam mata kuliah Desain Pemrograman Berorientasi Objek untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin.
*/

/* Import */
#include "header.hh"

// default constructor
Dosen::Dosen() : SivitasAkademik(){
    this->employeeID = "";
    this->lastEdu = "";
    this->skill = "";
}
Dosen::Dosen(string nik, string nama, string jenis_kelamin, string email, string asal_univ, string fakultas, string nip, string pend_terakhir, string keahlian) : SivitasAkademik(nik, nama, jenis_kelamin, email, asal_univ, fakultas) {
    this->employeeID = nip;
    this->lastEdu = pend_terakhir;
    this->skill = keahlian;
}

// setter and getter
void Dosen::setEmployeeID(string nip) {
    this->employeeID = nip;
}
void Dosen::setLastEdu(string pend_terakhir) {
    this->lastEdu = pend_terakhir;
}
void Dosen::setSkill(string keahlian) {
    this->skill = keahlian;
}
string Dosen::getEmployeeID() {
    return this->employeeID;
}
string Dosen::getLastEdu() {
    return this->lastEdu;
}
string Dosen::getSkill() {
    return this->skill;
}

// destructor
Dosen::~Dosen() {
}